Modules/Devices/src/Devices.Application/Tiers/Commands/CreateTier/Handler.cs

```diff
@@ -22,6 +23,10 @@ public Handler(ITiersRepository tierRepository, ILogger<Handler> logger, IEventB
 
     public async Task<CreateTierResponse> Handle(CreateTierCommand request, CancellationToken cancellationToken)
     {
+        var tierExists = await _tierRepository.ExistsWithName(request.Name, cancellationToken);
+        if (tierExists)
+            throw new OperationFailedException(ApplicationErrors.Devices.TierNameAlreadyExists());
```

Comment:
Can you throw an ApplicationException please? I don't remember why I even introduced the OperationFailedException. ;-)

Reply:
Sure. I followed the CreateClients handler for this validation, and the handler there uses this same OperationFailedException. Should I also change it there?

Reply:
That would be great, thank you